File: 2024/March/3-20 1669 Merge In Between Linked Lists/MergeBetween.java

class ListNode {
    int val;
    ListNode next;
    ListNode() {}
    ListNode(int val) { this.val = val; }
    ListNode(int val, ListNode next) { this.val = val; this.next = next; }
}
 
class Solution {
    public ListNode mergeInBetween(ListNode list1, int a, int b, ListNode list2) {
        ListNode list2Head = list2;
        ListNode list2Tail = list2;
        while(list2Tail.next != null) {
            list2Tail = list2Tail.next;
        }

        ListNode node = list1;
        ListNode prev = null;
        for(int i = 0; ; i++) {
            if(i == a) {
                prev.next = list2Head;
            }
            if(i == b) {
                list2Tail.next = node.next;
                break;
            }

            prev = node;
            node = node.next;
        }

        return list1;
    }
}

File: 2024/March/3-20 1669 Merge In Between Linked Lists/MergeBetween.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def mergeInBetween(self, list1: ListNode, a: int, b: int, list2: ListNode) -> ListNode:
        list2Head = list2
        list2Tail = list2
        while list2Tail.next:
            list2Tail = list2Tail.next

        node = list1
        prev = None
        i = 0
        while node:
            if i == a:
                prev.next = list2Head
            if i == b:
                list2Tail.next = node.next
                break
            
            prev = node
            node = node.next
            i += 1
        
        return list1